integration/projection_array_test.go

```diff
@@ -0,0 +1,210 @@
+package integration
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func TestProjectionQuerySlice(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup(t)
+	_, err := collection.InsertOne(ctx,
+		bson.D{{"_id", "array"}, {"value", bson.A{1, 2, 3, 4}}},
+	)
+	require.NoError(t, err)
+
+	type testCase struct {
```

Comment:
We don't use that structure anywhere outside the for loop below. Let's make embedded as in other tests.

Reply:
done